internal/service/appstream/find.go

```diff
@@ -92,3 +92,25 @@ func FindImageBuilderByName(ctx context.Context, conn *appstream.AppStream, name
 
 	return result, nil
 }
+
+// FindUserByUserNameAndAuthType Retrieve a appstream fleet by Username and authentication type
+func FindUserByUserNameAndAuthType(ctx context.Context, conn *appstream.AppStream, username, authType string) (*appstream.User, error) {
+	input := &appstream.DescribeUsersInput{
+		AuthenticationType: aws.String(authType),
+	}
+
+	var user *appstream.User
+	resp, err := conn.DescribeUsersWithContext(ctx, input)
+
+	if err != nil {
```

Comment:
If we move handling not-found into this function, then calling functions can be simpler. For instance, here, we can check the following before other errors

```go
if tfawserr.ErrCodeEquals(err, appstream.ErrCodeResourceNotFoundException) {
	return nil, &resource.NotFoundError{
		LastError:   err,
		LastRequest: input,
	}
}
```

And then, after looping over the results, if `user` is `nil`

```go
if user == nil {
	return nil, &resource.NotFoundError{
		Message:  "Empty result",
		LastRequest: input,
	}
}
```

Then, in code that calls this function, it can simply check for a not found error by using the function `tfresource.NotFound(err)`

See https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ds/find.go for an example